core/src/main/java/com/google/errorprone/bugpatterns/MissingTestCall.java

```diff
@@ -128,12 +127,7 @@ public Void visitMethodInvocation(MethodInvocationTree node, Void unused) {
 
   @Nullable
   private static ExpressionTree getUltimateReceiver(ExpressionTree tree) {
-    ExpressionTree receiver = getReceiver(tree);
-    while (receiver instanceof MemberSelectTree || receiver instanceof MethodInvocationTree) {
-      tree = receiver;
-      receiver = getReceiver(receiver);
-    }
-    return receiver;
+    return streamReceivers(tree).reduce((a, b) -> b).orElse(null);
```

Comment:
Guava has [`Streams.findLast`](https://guava.dev/releases/23.0/api/docs/com/google/common/collect/Streams.html#findLast-java.util.stream.Stream-) which is documented as:
> Equivalent to `stream.reduce((a, b) -> b)`, but may perform significantly better. This method's runtime will be between O(log n) and O(n), performing better on [efficiently splittable](http://gee.cs.oswego.edu/dl/html/StreamParallelGuidance.html) streams.
```suggestion
    return Streams.findLast(streamReceivers(tree)).orElse(null);
```

I guess `streamReceivers` is not "efficiently splittable", but arguably `findLast` more clearly states intent.

Reply:
Aha, thanks. I should have looked harder for a helper.